# บทที่ 2: ตัวแปร นิพจน์ และชนิดข้อมูลพื้นฐาน

ในบทนี้ เราจะเริ่มเรียนพื้นฐานภาษา Python อย่างเป็นระบบ โดยเน้น 3 เรื่องสำคัญ

1. **Variable** หรือตัวแปร  
2. **Expression** หรือนิพจน์  
3. **Basic Data Types** หรือชนิดข้อมูลพื้นฐาน

แนวคิดเหล่านี้เป็นพื้นฐานที่จำเป็นมากสำหรับงานข้อมูล เพราะก่อนที่เราจะวิเคราะห์ข้อมูลได้ เราต้องเข้าใจก่อนว่า Python เก็บค่า คำนวณค่า และแยกประเภทของข้อมูลอย่างไร

สำหรับผู้ที่คุ้นเคยกับ Excel ให้ลองนึกว่า

- ตัวแปร คล้ายกับการตั้งชื่อให้ค่าหรือผลลัพธ์บางอย่าง
- นิพจน์ คล้ายกับสูตรคำนวณ
- ชนิดข้อมูล คล้ายกับประเภทของข้อมูลใน cell เช่น ตัวเลข ข้อความ วันที่ หรือค่า TRUE/FALSE

ในบทนี้ เราจะยังไม่เข้าสู่การวิเคราะห์ข้อมูลตารางเต็มรูปแบบ แต่จะวางพื้นฐานที่จำเป็นก่อนเข้าสู่บทต่อ ๆ ไป

## ผลการเรียนรู้ที่คาดหวัง

เมื่อเรียนจบบทนี้ ผู้เรียนจะสามารถ

1. สร้างและใช้งานตัวแปรใน Python ได้
2. ตั้งชื่อตัวแปรได้เหมาะสมและอ่านเข้าใจง่าย
3. ใช้นิพจน์เพื่อคำนวณค่าพื้นฐานได้
4. แยกชนิดข้อมูลพื้นฐาน ได้แก่ integer, float, string และ boolean ได้
5. ตรวจสอบชนิดข้อมูลด้วย `type()` ได้
6. แปลงชนิดข้อมูลพื้นฐานได้ในกรณีที่จำเป็น
7. ใช้ f-string เพื่อสร้างข้อความสรุปผลลัพธ์จากตัวแปรได้
8. เปลี่ยนสูตรหรือการคำนวณที่เคยทำใน Excel ให้เป็น Python expression ได้

## ลำดับเนื้อหา

บทเรียนนี้ประกอบด้วยหัวข้อต่อไปนี้

1. ตัวแปรคืออะไร
2. การตั้งชื่อตัวแปร
3. นิพจน์และตัวดำเนินการพื้นฐาน
4. ลำดับการคำนวณ
5. ชนิดข้อมูลพื้นฐาน
6. การตรวจสอบชนิดข้อมูล
7. การแปลงชนิดข้อมูล
8. การสร้างข้อความจากตัวแปรด้วย f-string
9. ข้อผิดพลาดที่พบบ่อย
10. แบบฝึกหัดท้ายบท

## 1. ตัวแปรคืออะไร

**ตัวแปร** หรือ **variable** คือชื่อที่เราใช้เก็บค่าไว้ใน Python

เมื่อต้องใช้ค่านั้นซ้ำ เราสามารถเรียกใช้ผ่านชื่อตัวแปรได้ แทนที่จะเขียนค่าซ้ำหลายครั้ง

ตัวอย่างเช่น หากเราต้องการเก็บยอดขายของสินค้า เราอาจสร้างตัวแปรชื่อ `sales`

In [2]:
sales = 15000
sales

15000

จากตัวอย่างด้านบน

```python
sales = 15000
```

หมายความว่า เรากำหนดค่า 15000 ให้กับตัวแปรชื่อ sales

เครื่องหมาย = ใน Python หมายถึง “กำหนดค่าให้ตัวแปร” ไม่ใช่เครื่องหมายเท่ากับแบบสมการทางคณิตศาสตร์

## ตัวแปรช่วยให้งานอ่านง่ายขึ้น

เปรียบเทียบโค้ด 2 แบบด้านล่าง

แบบที่ 1: ใช้ตัวเลขโดยตรง

In [3]:
15000 * 0.07

1050.0

แบบที่ 2: ใช้ชื่อตัวแปร

In [4]:
sales = 15000
vat_rate = 0.07

vat_amount = sales * vat_rate
vat_amount

1050.0

โค้ดแบบที่ 2 อ่านเข้าใจง่ายกว่า เพราะเห็นความหมายของแต่ละค่าอย่างชัดเจน

- `sales` คือยอดขาย
- `vat_rate` คืออัตราภาษี
- `vat_amount` คือจำนวนภาษี

ในการทำงานข้อมูลจริง การตั้งชื่อตัวแปรให้มีความหมายจะช่วยลดความสับสนและลดข้อผิดพลาดได้มาก

## 2. การตั้งชื่อตัวแปร

การตั้งชื่อตัวแปรใน Python มีกฎพื้นฐานดังนี้

1. ใช้ตัวอักษร ตัวเลข และ underscore `_` ได้
2. ห้ามขึ้นต้นด้วยตัวเลข
3. ห้ามมีช่องว่าง
4. ตัวพิมพ์เล็กและตัวพิมพ์ใหญ่ถือว่าเป็นคนละตัว
5. ไม่ควรใช้คำสงวนของ Python เช่น `if`, `for`, `class`, `True`, `False`

แนวทางที่แนะนำคือใช้ตัวอักษรพิมพ์เล็ก และคั่นคำด้วย underscore เช่น

```python
total_sales
average_price
customer_count
```

In [5]:
total_sales = 45000
average_price = 250
customer_count = 180

## ตัวอย่างชื่อตัวแปรที่ดีและไม่ดี

| ตัวแปร | ความเห็น |
|---|---|
| `sales` | ดี อ่านง่าย |
| `total_sales` | ดี สื่อความหมายชัดเจน |
| `sales_2026` | ใช้ได้ |
| `x` | ใช้ได้ แต่ไม่สื่อความหมาย |
| `total sales` | ใช้ไม่ได้ เพราะมีช่องว่าง |
| `2026_sales` | ใช้ไม่ได้ เพราะขึ้นต้นด้วยตัวเลข |
| `TotalSales` | ใช้ได้ แต่ไม่ใช่ style ที่นิยมในตัวอย่างพื้นฐานของ Python |

## 3. นิพจน์และตัวดำเนินการพื้นฐาน

**นิพจน์** หรือ **expression** คือส่วนของโค้ดที่ Python สามารถประมวลผลแล้วได้ค่าออกมา

ตัวอย่างเช่น

```python
100 + 50
sales * vat_rate
(total_sales / customer_count)
```

ตัวดำเนินการพื้นฐานที่ใช้บ่อย ได้แก่

| ตัวดำเนินการ | ความหมาย | ตัวอย่าง |
|---|---|---|
|+ | บวก | 10 + 5 |
|- | ลบ | 10 - 5 |
|* | คูณ |	10 * 5 |
|/ | หาร |	10 / 5 |
|** | ยกกำลัง |	10 ** 2 |
|% | เศษจากการหาร |	10 % 3 |
|// | หารแบบปัดเศษลง |	10 // 3 |

In [6]:
revenue = 120000
cost = 75000

profit = revenue - cost
profit

45000

In [7]:
profit_margin = profit / revenue
profit_margin

0.375

ใน Excel เราอาจเขียนสูตรหา profit margin ว่า

```excel
=(Revenue-Cost)/Revenue
```

ใน Python เราสามารถแยกเป็นขั้นตอนโดยใช้ตัวแปร ทำให้โค้ดอ่านง่ายขึ้น

In [8]:
profit = revenue - cost
profit_margin = profit / revenue
profit_margin

0.375

## 4. ลำดับการคำนวณ

Python ใช้ลำดับการคำนวณคล้ายคณิตศาสตร์ทั่วไป

ลำดับโดยสรุปคือ

1. วงเล็บ `()`
2. ยกกำลัง `**`
3. คูณ หาร หารปัดเศษ และเศษจากการหาร `*`, `/`, `//`, `%`
4. บวกและลบ `+`, `-`

เมื่อไม่แน่ใจ ควรใช้วงเล็บเพื่อให้โค้ดอ่านง่ายและลดความสับสน

In [9]:
result_1 = 10 + 5 * 2
result_1

20

In [10]:
result_2 = (10 + 5) * 2
result_2

30

สองตัวอย่างด้านบนให้ผลลัพธ์ต่างกัน เพราะวงเล็บเปลี่ยนลำดับการคำนวณ

ในการเขียนสูตรวิเคราะห์ข้อมูล ควรใช้วงเล็บเมื่อสูตรเริ่มซับซ้อน เช่น

```python
gross_profit_margin = (revenue - cost) / revenue
```

## 5. ชนิดข้อมูลพื้นฐาน

Python แยกประเภทของข้อมูลออกเป็นหลายชนิด  
ในบทนี้เราจะเริ่มจากชนิดข้อมูลพื้นฐานที่ใช้บ่อยในงานข้อมูล

| ชนิดข้อมูล | ชื่อใน Python | ตัวอย่าง |
|---|---|---|
| จำนวนเต็ม | `int` | `100`, `2500`, `-5` |
| จำนวนทศนิยม | `float` | `3.14`, `0.07`, `125.50` |
| ข้อความ | `str` | `"Bangkok"`, `"Product A"` |
| ค่าจริง/เท็จ | `bool` | `True`, `False` |

## Integer: จำนวนเต็ม

`int` ใช้เก็บจำนวนเต็ม เช่น จำนวนลูกค้า จำนวนรายการ หรือจำนวนสินค้า

In [11]:
order_count = 125
order_count

125

## Float: จำนวนทศนิยม

`float` ใช้เก็บจำนวนที่มีทศนิยม เช่น อัตราภาษี อัตราการเติบโต หรือค่าเฉลี่ย

In [12]:
growth_rate = 0.125
growth_rate

0.125

## String: ข้อความ

`str` ใช้เก็บข้อความ เช่น ชื่อสินค้า ชื่อจังหวัด รหัสลูกค้า หรือหมวดหมู่ข้อมูล

ข้อความใน Python ต้องอยู่ในเครื่องหมาย quote เช่น `" "` หรือ `' '`

In [13]:
product_name = "Keyboard"
region = "Bangkok"

product_name

'Keyboard'

In [14]:
region

'Bangkok'

ข้อควรระวัง: ตัวเลขที่อยู่ใน quote จะถูกมองเป็นข้อความ ไม่ใช่ตัวเลข

In [15]:
customer_id = "1001"
customer_id

'1001'

ในตัวอย่างนี้ `customer_id` แม้จะดูเหมือนตัวเลข แต่ควรเก็บเป็นข้อความ เพราะเราไม่ได้ต้องการนำรหัสลูกค้าไปบวก ลบ คูณ หรือหาร

## Boolean: ค่าจริงหรือเท็จ

`bool` ใช้เก็บค่าที่เป็นจริงหรือเท็จ

ค่าของ boolean มีเพียง 2 ค่า

```python
True
False
```

Boolean มักใช้กับเงื่อนไข เช่น

- ลูกค้าคนนี้เป็นสมาชิกหรือไม่
- รายการนี้มียอดขายมากกว่าเป้าหมายหรือไม่
- ข้อมูลแถวนี้มี missing value หรือไม่

In [16]:
is_member = True
is_member

True

In [17]:
sales = 15000
target = 10000

is_above_target = sales > target
is_above_target

True

ตัวอย่างด้านบนใช้เครื่องหมาย `>` เพื่อตรวจสอบว่า `sales` มากกว่า `target` หรือไม่

ผลลัพธ์ที่ได้คือ boolean

- ถ้าเงื่อนไขเป็นจริง จะได้ `True`
- ถ้าเงื่อนไขเป็นเท็จ จะได้ `False`

## 6. การตรวจสอบชนิดข้อมูล

เราสามารถใช้ `type()` เพื่อตรวจสอบว่าค่าหรือตัวแปรเป็นข้อมูลชนิดใด

In [18]:
type(order_count)

int

In [19]:
type(growth_rate)

float

In [20]:
type(is_member)

bool

การตรวจสอบชนิดข้อมูลเป็นทักษะสำคัญมากในงานข้อมูล เพราะหลายครั้งข้อมูลที่ดูเหมือนตัวเลข อาจถูกอ่านเข้ามาเป็นข้อความ

เช่น

```python
"1000"
```

ไม่เหมือนกับ

```python
1000
```

ค่าหนึ่งเป็นข้อความ อีกค่าหนึ่งเป็นตัวเลข

In [21]:
amount_text = "1000"
amount_number = 1000

type(amount_text)

str

In [22]:
type(amount_number)

int

## 7. การแปลงชนิดข้อมูล

บางครั้งเราจำเป็นต้องแปลงชนิดข้อมูลก่อนนำไปคำนวณ

ฟังก์ชันที่ใช้บ่อย ได้แก่

| ฟังก์ชัน | ใช้ทำอะไร |
|---|---|
| `int()` | แปลงเป็นจำนวนเต็ม |
| `float()` | แปลงเป็นจำนวนทศนิยม |
| `str()` | แปลงเป็นข้อความ |
| `bool()` | แปลงเป็นค่าจริง/เท็จ |

In [23]:
amount_text = "12500"

amount_number = int(amount_text)

amount_number

12500

In [24]:
type(amount_number)

int

In [25]:
price_text = "125.50"

price_number = float(price_text)

price_number

125.5

ถ้าข้อความไม่ใช่รูปแบบตัวเลขที่ถูกต้อง Python จะแปลงไม่ได้

In [26]:
# ลองรันเพื่อดู error ที่เกิดขึ้น

int("Bangkok")

ValueError: invalid literal for int() with base 10: 'Bangkok'

ในงานข้อมูลจริง ปัญหาชนิดข้อมูลเกิดขึ้นบ่อย เช่น

- ตัวเลขถูกอ่านมาเป็นข้อความ
- วันที่ถูกอ่านมาเป็นข้อความ
- รหัสที่ขึ้นต้นด้วยเลขศูนย์ถูกมองเป็นตัวเลข
- ช่องว่างหรือ comma ทำให้แปลงเป็นตัวเลขไม่ได้

ตัวอย่างเช่น `"12,500"` ไม่สามารถแปลงด้วย `int()` ตรง ๆ ได้ เพราะมี comma อยู่ในข้อความ

In [27]:
sales_text = "12,500"

clean_sales_text = sales_text.replace(",", "")
sales_number = int(clean_sales_text)

sales_number

12500

โค้ดด้านบนใช้แนวคิดง่าย ๆ คือ

1. เริ่มจากข้อความ `"12,500"`
2. ลบ comma ออกด้วย `.replace(",", "")`
3. แปลงข้อความที่เหลือเป็นตัวเลขด้วย `int()`

รายละเอียดของ string method จะเรียนลึกขึ้นในบทถัดไป  
ตอนนี้ให้เห็นก่อนว่าชนิดข้อมูลมีผลต่อการคำนวณ

## 8. การสร้างข้อความจากตัวแปรด้วย f-string

ในการวิเคราะห์ข้อมูล เรามักต้องการสร้างข้อความสรุปผลลัพธ์ เช่น

- ยอดขายรวมเท่ากับ 125,000 บาท
- อัตรากำไรเท่ากับ 37.5%
- จังหวัดที่เลือกคือ Bangkok

Python มีวิธีสร้างข้อความจากตัวแปรที่อ่านง่าย เรียกว่า **f-string**

In [28]:
total_sales = 125000
summary = f"ยอดขายรวมเท่ากับ {total_sales} บาท"

summary

'ยอดขายรวมเท่ากับ 125000 บาท'

การเขียน f-string ทำได้โดยใส่ตัวอักษร `f` หน้า quote แล้วใส่ชื่อตัวแปรไว้ใน `{}`

```python
f"ยอดขายรวมเท่ากับ {total_sales} บาท"
```

เราสามารถใส่นิพจน์ไว้ใน `{}` ได้ด้วย

In [29]:
revenue = 120000
cost = 75000

profit = revenue - cost
profit_margin = profit / revenue

summary = f"กำไรเท่ากับ {profit} บาท และอัตรากำไรเท่ากับ {profit_margin:.2%}"

summary

'กำไรเท่ากับ 45000 บาท และอัตรากำไรเท่ากับ 37.50%'

ในตัวอย่างด้านบน

```python
{profit_margin:.2%}
```

หมายถึง แสดงค่า `profit_margin` เป็นเปอร์เซ็นต์ โดยมีทศนิยม 2 ตำแหน่ง

การจัดรูปแบบลักษณะนี้มีประโยชน์มากเวลาสร้างข้อความสรุปผลการวิเคราะห์

## 9. ข้อผิดพลาดที่พบบ่อย

### 1. ใช้ตัวแปรก่อนสร้าง

ถ้าเรียกใช้ตัวแปรที่ยังไม่เคยกำหนดค่า Python จะเกิด error

In [ ]:
total_profit

NameError: name 'total_profit' is not defined

### 2. สะกดชื่อตัวแปรไม่ตรงกัน

Python แยกตัวพิมพ์เล็กและตัวพิมพ์ใหญ่

`total_sales` และ `Total_sales` ถือว่าเป็นคนละตัวแปร

In [ ]:
total_sales = 10000

Total_sales

NameError: name 'Total_sales' is not defined

### 3. นำข้อความไปคำนวณเหมือนตัวเลข

ถ้าตัวเลขถูกเก็บเป็นข้อความ อาจคำนวณไม่ได้ตามที่ต้องการ

In [30]:
sales_a = "1000"
sales_b = "2000"

sales_a + sales_b

'10002000'

ผลลัพธ์จาก Cell ด้านบนคือการนำข้อความมาต่อกัน ไม่ใช่การบวกตัวเลข

ถ้าต้องการบวกเป็นตัวเลข ต้องแปลงชนิดข้อมูลก่อน

In [31]:
int(sales_a) + int(sales_b)

3000

### 4. ใส่ quote ผิดตำแหน่ง

ข้อความต้องอยู่ใน quote แต่ตัวแปรไม่ควรอยู่ใน quote หากต้องการใช้ค่าของตัวแปร

In [32]:
region = "Bangkok"

message_1 = "region"
message_2 = region

message_1

'region'

In [33]:
message_2

'Bangkok'

`message_1` ได้ข้อความว่า `"region"` โดยตรง  
แต่ `message_2` ได้ค่าที่เก็บอยู่ในตัวแปร `region`

# แบบฝึกหัดท้ายบท  

ในแบบฝึกหัดนี้ ให้ฝึกใช้ตัวแปร นิพจน์ และชนิดข้อมูลพื้นฐานใน Python ผ่านบริบทของข้อมูลเชิงสำรวจในงานสังคมสงเคราะห์

เป้าหมายของแบบฝึกหัดนี้ไม่ใช่แค่การคำนวณให้ได้คำตอบที่ถูกต้อง แต่เพื่อฝึกคิดว่า

- ควรตั้งชื่อตัวแปรอย่างไรให้อ่านเข้าใจง่าย
- ข้อมูลแต่ละประเภทควรเก็บเป็นชนิดข้อมูลใด
- สูตรใน Excel สามารถเขียนเป็น Python expression ได้อย่างไร
- ผลลัพธ์จากการคำนวณสามารถนำไปเขียนเป็นข้อความสรุปได้อย่างไร

## แบบฝึกหัดที่ 1: ตั้งชื่อตัวแปรจากข้อมูลแบบสำรวจ

สมมติว่าในการสำรวจครัวเรือนเปราะบาง มีข้อมูลเบื้องต้นดังนี้

| ความหมาย | ตัวอย่างค่า |
|---|---:|
| จำนวนผู้ตอบแบบสำรวจทั้งหมด | 180 |
| จำนวนผู้ตอบที่เป็นผู้สูงอายุ | 72 |
| คะแนนเฉลี่ยความเปราะบาง | 3.8 |
| มีการเก็บข้อมูลครบถ้วนหรือไม่ | True |

ให้สร้างตัวแปรใน Python เพื่อเก็บค่าทั้ง 4 รายการนี้

เงื่อนไข:

- ใช้ชื่อตัวแปรภาษาอังกฤษ
- ใช้ตัวพิมพ์เล็ก
- คั่นคำด้วย underscore
- ชื่อตัวแปรควรสื่อความหมาย
- เลือกชนิดข้อมูลให้เหมาะสมกับค่าที่เก็บ

In [34]:
# เขียนคำตอบของคุณใน Cell นี้

## แบบฝึกหัดที่ 2: แปลงสูตร Excel เป็น Python Expression

สมมติว่ามีข้อมูลจำนวนผู้ตอบแบบสำรวจทั้งหมด และจำนวนผู้ตอบที่ระบุว่าต้องการความช่วยเหลือเร่งด่วน

```python
total_respondents = 180
urgent_need_count = 54
```

ใน Excel เราอาจคำนวณสัดส่วนผู้ที่ต้องการความช่วยเหลือเร่งด่วนด้วยสูตร

```excel
=UrgentNeedCount/TotalRespondents
```

ให้ใช้ Python คำนวณค่า `urgent_need_ratio`

In [1]:
total_respondents = 180
urgent_need_count = 54

In [36]:
# เขียนคำตอบของคุณใน Cell นี้

## แบบฝึกหัดที่ 3: คำนวณร้อยละของผู้ตอบแบบสำรวจ

จากข้อมูลในแบบฝึกหัดก่อนหน้า

```python
total_respondents = 180
urgent_need_count = 54
```

ให้คำนวณร้อยละของผู้ตอบที่ต้องการความช่วยเหลือเร่งด่วน

ใน Excel เราอาจใช้สูตร

```excel
=(UrgentNeedCount/TotalRespondents)*100
```

ให้สร้างตัวแปรชื่อ `urgent_need_percent`

จากนั้นใช้ f-string แสดงข้อความสรุป เช่น

```text
ร้อยละของผู้ตอบที่ต้องการความช่วยเหลือเร่งด่วนเท่ากับ 30.00%
```

In [2]:
total_respondents = 180
urgent_need_count = 54

In [38]:
# เขียนคำตอบของคุณใน Cell นี้

## แบบฝึกหัดที่ 4: ตรวจสอบชนิดข้อมูลจากแบบสำรวจ

พิจารณาค่าต่อไปนี้

```python
respondent_id = "000127"
age = 68
vulnerability_score = 4.25
has_disability = True
```

ให้ใช้ `type()` ตรวจสอบชนิดข้อมูลของแต่ละตัวแปร

จากนั้นตอบคำถาม:

1. ตัวแปรใดเป็นข้อความ หรือ string
2. ตัวแปรใดเป็นจำนวนเต็ม หรือ integer
3. ตัวแปรใดเป็นจำนวนทศนิยม หรือ float
4. ตัวแปรใดเป็น boolean
5. เพราะเหตุใด `respondent_id` จึงควรเป็นข้อความ แม้จะประกอบด้วยตัวเลข

In [3]:
respondent_id = "000127"
age = 68
vulnerability_score = 4.25
has_disability = True

In [40]:
# เขียนคำตอบของคุณใน Cell นี้

## แบบฝึกหัดที่ 5: แปลงข้อความเป็นตัวเลขก่อนคำนวณ

สมมติว่าข้อมูลจำนวนผู้ตอบแบบสำรวจถูกคัดลอกจาก Excel หรือ Google Forms แล้วอยู่ในรูปแบบข้อความ

```python
respondents_area_a = "65"
respondents_area_b = "48"
```

ถ้านำสองค่านี้มาบวกกันโดยตรง Python จะต่อข้อความเข้าด้วยกัน ไม่ใช่บวกตัวเลข

ให้ทำตามขั้นตอนต่อไปนี้

1. แปลง `respondents_area_a` เป็นจำนวนเต็ม
2. แปลง `respondents_area_b` เป็นจำนวนเต็ม
3. คำนวณจำนวนผู้ตอบแบบสำรวจรวมจากทั้งสองพื้นที่
4. เก็บผลลัพธ์ไว้ในตัวแปรชื่อ `total_respondents`

In [4]:
respondents_area_a = "65"
respondents_area_b = "48"

In [42]:
# เขียนคำตอบของคุณใน Cell นี้

## แบบฝึกหัดที่ 6: จัดการตัวเลขที่มี comma

สมมติว่าจำนวนงบประมาณช่วยเหลือครัวเรือนจากไฟล์ Excel ถูก copy มาเป็นข้อความที่มี comma

```python
support_budget_text = "1,250,000"
```

ให้ทำตามขั้นตอนต่อไปนี้

1. ลบ comma ออกจากข้อความ
2. แปลงข้อความเป็นจำนวนเต็ม
3. เก็บผลลัพธ์ไว้ในตัวแปรชื่อ `support_budget`
4. สร้างข้อความสรุปด้วย f-string ว่า

```text
งบประมาณช่วยเหลือทั้งหมดเท่ากับ 1250000 บาท
```

In [5]:
support_budget_text = "1,250,000"

In [44]:
# เขียนคำตอบของคุณใน Cell นี้

## แบบฝึกหัดที่ 7: คำนวณคะแนนเฉลี่ยจากแบบสอบถาม

สมมติว่าแบบสอบถามประเมินความต้องการความช่วยเหลือมี 3 ด้าน โดยแต่ละด้านให้คะแนนตั้งแต่ 1 ถึง 5

```python
income_need_score = 4.2
housing_need_score = 3.7
mental_health_need_score = 4.5
```

ให้คำนวณคะแนนเฉลี่ยรวมของทั้ง 3 ด้าน

ใน Excel เราอาจใช้สูตร

```excel
=AVERAGE(IncomeNeedScore,HousingNeedScore,MentalHealthNeedScore)
```

ให้สร้างตัวแปรชื่อ `average_need_score`

จากนั้นใช้ f-string แสดงข้อความสรุป เช่น

```text
คะแนนเฉลี่ยความต้องการความช่วยเหลือเท่ากับ 4.13 คะแนน
```

In [6]:
income_need_score = 4.2
housing_need_score = 3.7
mental_health_need_score = 4.5

In [46]:
# เขียนคำตอบของคุณใน Cell นี้

## แบบฝึกหัดที่ 8: ตรวจสอบว่าคะแนนถึงเกณฑ์เฝ้าระวังหรือไม่

สมมติว่านักสังคมสงเคราะห์กำหนดว่า

- หากคะแนนความเปราะบางตั้งแต่ 4.0 ขึ้นไป ถือว่าอยู่ในกลุ่มที่ควรติดตามเป็นพิเศษ

มีข้อมูลดังนี้

```python
vulnerability_score = 4.25
risk_threshold = 4.0
```

ใน Excel เราอาจเขียนสูตรเปรียบเทียบ เช่น

```excel
=VulnerabilityScore>=RiskThreshold
```

ให้ใช้ Python ตรวจสอบว่าคะแนนความเปราะบางถึงเกณฑ์เฝ้าระวังหรือไม่

ให้สร้างตัวแปรชื่อ `is_high_risk` และเก็บผลลัพธ์เป็นค่า boolean

In [7]:
vulnerability_score = 4.25
risk_threshold = 4.0

In [48]:
# เขียนคำตอบของคุณใน Cell นี้

## แบบฝึกหัดที่ 9: สร้างข้อความสรุปผลจากตัวแปรด้วย f-string

สมมติว่ามีข้อมูลสรุปจากแบบสำรวจในพื้นที่หนึ่งดังนี้

```python
area_name = "Bangkok"
total_respondents = 180
urgent_need_count = 54
average_need_score = 4.13
```

ให้ใช้ f-string เพื่อสร้างข้อความสรุปผลเบื้องต้นในรูปแบบต่อไปนี้

```text
พื้นที่ Bangkok มีผู้ตอบแบบสำรวจทั้งหมด 180 คน โดยมีผู้ที่ต้องการความช่วยเหลือเร่งด่วน 54 คน และมีคะแนนเฉลี่ยความต้องการความช่วยเหลือเท่ากับ 4.13 คะแนน
```

ให้เก็บข้อความไว้ในตัวแปรชื่อ `summary_text` แล้วแสดงผลลัพธ์ออกมา

In [8]:
area_name = "Bangkok"
total_respondents = 180
urgent_need_count = 54
average_need_score = 4.13

In [9]:
# เขียนคำตอบของคุณใน Cell นี้

## แบบฝึกหัดที่ 10: รวมตัวแปร นิพจน์ ชนิดข้อมูล และ f-string เข้าด้วยกัน

สมมติว่าคุณได้รับข้อมูลสรุปจากแบบสำรวจครัวเรือนในพื้นที่หนึ่ง

```python
area_name = "Chiang Mai"
total_households = "240"
households_with_children = "96"
average_vulnerability_score = 3.85
high_risk_threshold = 4.0
```

ให้เขียน Python เพื่อทำงานต่อไปนี้

1. แปลง `total_households` เป็นจำนวนเต็ม
2. แปลง `households_with_children` เป็นจำนวนเต็ม
3. คำนวณสัดส่วนครัวเรือนที่มีเด็ก
4. คำนวณร้อยละครัวเรือนที่มีเด็ก
5. ตรวจสอบว่า `average_vulnerability_score` ถึงเกณฑ์ความเสี่ยงสูงหรือไม่ โดยเปรียบเทียบกับ `high_risk_threshold`
6. สร้างข้อความสรุปด้วย f-string ในรูปแบบใกล้เคียงกับตัวอย่างนี้

>พื้นที่ Chiang Mai มีข้อมูลทั้งหมด 240 ครัวเรือน โดยมีครัวเรือนที่มีเด็ก 96 ครัวเรือน คิดเป็น 40.00% >และสถานะความเสี่ยงสูงคือ False


In [10]:
# เขียนคำตอบของคุณใน Cell นี้